src/core/Totp.cpp

```diff
@@ -180,7 +181,7 @@ QString Totp::writeSettings(const QSharedPointer<Totp::Settings>& settings,
         auto urlstring = QString("otpauth://totp/%1:%2?secret=%3&period=%4&digits=%5&issuer=%1")
                              .arg(title.isEmpty() ? "KeePassXC" : QString(QUrl::toPercentEncoding(title)),
                                   username.isEmpty() ? "none" : QString(QUrl::toPercentEncoding(username)),
-                                  QString(QUrl::toPercentEncoding(Base32::sanitizeInput(settings->key.toLatin1()))),
+                                  QString(Base32::sanitizeInput(settings->key.toLatin1())).remove(QRegularExpression("=+$")),
```

Comment:
It would probably make sense to add the padding option inside the Base32 class, without removing any characters afterwards.